src/test/java/com/emc/object/s3/S3MetadataSearchTest.java

```diff
@@ -237,7 +381,99 @@ public void testListObjectsWithEncoding() {
+                    .withPrefix("prefix1/");
+            result = client.queryObjects(request);
+
+            Assert.assertEquals(0, result.getObjects().size());
+             */
+
+            // list the bad key
+            /* Blocked by STORAGE-30527. Uncomment after fixed.*/
+            /*
+            request = new QueryObjectsRequest(bucketName).withEncodingType(EncodingType.url)
+                    .withQuery("(x-amz-meta-field-valid=='true') and (x-amz-meta-index-field>'')")
+                    .withPrefix("prefix/");
+            result = client.queryObjects(request);
+
+            Assert.assertEquals(2, result.getObjects().size());
+            Assert.assertEquals(badKey, RestUtil.urlDecode(result.getObjects().get(0).getObjectName()));
+            Assert.assertEquals(goodKey, RestUtil.urlDecode(result.getObjects().get(1).getObjectName()));
+             */
+
+            // list a good field, with bad field results
+            request = new QueryObjectsRequest(bucketName).withEncodingType(EncodingType.url)
+                    .withQuery("x-amz-meta-field-valid=='false'").withPrefix("prefix/");
+            result = client.queryObjects(request);
+
+            Assert.assertEquals(1, result.getObjects().size());
+            Assert.assertEquals(badField, RestUtil.urlDecode(result.getObjects().get(0).getObjectName()));
+
+            // list a bad field
+            request = new QueryObjectsRequest(bucketName).withEncodingType(EncodingType.url)
+                    .withQuery("x-amz-meta-index-field=='" + RestUtil.urlEncode(badFieldValue) + "'")
+                    .withPrefix("prefix/");
+            result = client.queryObjects(request);
+
+            Assert.assertEquals(1, result.getObjects().size());
+            Assert.assertEquals(badField, RestUtil.urlDecode(result.getObjects().get(0).getObjectName()));
+
+            List<QueryMetadata> queryMds = result.getObjects().get(0).getQueryMds();
+
+            //SYSMD and USERMD
+            Assert.assertEquals(2, queryMds.size());
+            QueryMetadata usermd = null;
+            for (QueryMetadata m : queryMds) {
+                switch (m.getType()) {
+                    case USERMD:
+                        usermd = m;
+                        break;
+                }
+            }
+            Assert.assertNotNull(usermd);
+            //badFieldValue has to be stored in url encoded format. Limit by SDK-553, user application needs to record encoded or not.
+            Assert.assertEquals(badFieldValue, RestUtil.urlDecode(usermd.getMdMap().get("x-amz-meta-index-field")));
+            Assert.assertEquals("false", RestUtil.urlDecode(usermd.getMdMap().get("x-amz-meta-field-valid")));
+            Assert.assertEquals("true", RestUtil.urlDecode(usermd.getMdMap().get("x-amz-meta-key-valid")));
+        } finally {
+            client.deleteObject(getTestBucket(), badKey);
```

Comment:
All objects created under testbucket are supposed to be cleaned after test is completed, this is not needed. and you could just use "throws Exception".

Reply:
This is from previous test cases, have removed all places with unneccesarry try/finally to delete objects in this test file.